chunks 1-2: support arbitrary second .Bx argument (to be pushed upstream)
chunks 3-4: support .Lb in the synopsis (to be pushed upstream)

Index: tmac/mdoc/doc-syms
--- tmac/mdoc/doc-syms.orig
+++ tmac/mdoc/doc-syms
@@ -86,6 +86,7 @@
 .\" NS local variable:
 .\" NS   doc-str-Bx
 .\" NS   doc-str-Bx1
+.\" NS   doc-str-Bx2
 .\" NS   doc-str-Bx-XXX
 .\" NS
 .\" NS width register 'Bx' defined in doc-common
@@ -132,9 +133,16 @@
 .            ie (\n[doc-type\n[doc-arg-ptr]] == 2) \{\
 .              ie d doc-str-Bx-\*[doc-arg\n[doc-arg-ptr]] \
 .                as doc-str-Bx1 "\*[doc-str-Bx-\*[doc-arg\n[doc-arg-ptr]]]
-.              el \
-.                nr doc-arg-ptr -1
-.            \}
+.              el \{\
+.                as doc-str-Bx1 "\-
+.                ds doc-str-Bx2 "\*[doc-arg\n[doc-arg-ptr]]
+.                substring doc-str-Bx2 0 0
+.                stringup doc-str-Bx2
+.                as doc-str-Bx1 "\*[doc-str-Bx2]
+.                ds doc-str-Bx2 "\*[doc-arg\n[doc-arg-ptr]]
+.                substring doc-str-Bx2 1
+.                as doc-str-Bx1 "\*[doc-str-Bx2]
+.            \}\}
 .            el \
 .              nr doc-arg-ptr -1
 .          \}
@@ -782,6 +790,7 @@
 .\" NS   doc-reg-Lb
 .\" NS   doc-str-Lb
 .\" NS   doc-str-Lb1
+.\" NS   doc-str-Lb2
 .\" NS   doc-str-Lb-XXX
 .\" NS
 .\" NS width register 'Lb' defined in doc-common
@@ -877,6 +886,28 @@
 .  ie (\n[doc-arg-count] >= \n[doc-arg-ptr]) \{\
 .    nr doc-curr-font \n[.f]
 .    ds doc-str-Lb \f[\n[doc-curr-font]]
+.
+.    if \n[doc-in-synopsis-section] \{\
+.      while (\n[doc-arg-ptr] <= \n[doc-arg-count]) \{\
+.        ds doc-str-Lb1 \*[doc-arg\n[doc-arg-ptr]]
+.        ds doc-str-Lb2 \*[doc-str-Lb1]
+.        substring doc-str-Lb2 0 2
+.        if '\*[doc-str-Lb2]'lib' \
+.          substring doc-str-Lb1 3
+.        ds doc-str-Lb1 \-l\*[doc-str-Lb1]
+.        if (\n[doc-arg-ptr] == 1) \
+.          ds doc-str-Lb1 "/* \*[doc-str-Lb1]
+.        if (\n[doc-arg-ptr] == \n[doc-arg-count]) \
+.          ds doc-str-Lb1 "\*[doc-str-Lb1] */
+.        ds doc-arg\n[doc-arg-ptr] "\*[doc-str-Lb1]
+.        nr doc-arg-ptr +1
+.      \}
+.      nr doc-arg-ptr 1
+.      br
+.      doc-print-recursive
+.      br
+.      return
+.    \}
 .
 .    ie d doc-str-Lb-\*[doc-arg\n[doc-arg-ptr]] \
 .      ds doc-str-Lb1 "\*[doc-str-Lb-\*[doc-arg\n[doc-arg-ptr]]]
